src/main/java/com/microsoft/sqlserver/jdbc/SQLServerConnection.java

```diff
@@ -4636,14 +4636,42 @@ public void setHoldability(int holdability) throws SQLServerException {
     }
 
     public int getNetworkTimeout() throws SQLException {
-        // this operation is not supported
-        throw new SQLFeatureNotSupportedException(SQLServerException.getErrString("R_notSupported"));
+        loggerExternal.entering(getClassNameLogging(), "getNetworkTimeout");
+
+        checkClosed();
+
+        int timeout = 0;
+        try {
+            timeout = tdsChannel.getNetworkTimeout();
+        }
+        catch (IOException ioe) {
+            terminate(SQLServerException.DRIVER_ERROR_IO_FAILED, ioe.getMessage(), ioe);
+        }
+
+        loggerExternal.exiting(getClassNameLogging(), "getNetworkTimeout");
+        return timeout;
     }
 
     public void setNetworkTimeout(Executor executor,
             int timeout) throws SQLException {
-        // this operation is not supported
-        throw new SQLFeatureNotSupportedException(SQLServerException.getErrString("R_notSupported"));
+        loggerExternal.entering(getClassNameLogging(), "setNetworkTimeout");
```

Comment:
we may also want to log the value of timeout by doing something like `loggerExternal.entering(getClassNameLogging(), "setNetworkTimeout", timeout);`